### PR TITLE
Feat: Client Driven Art Pass

### PR DESCRIPTION
**Pull request by jilfranco-unity**
### Description

A PR for the first art pass on the client driven sample! I've used the starter assets as well as some custom assets & shaders to dress up our level. Final animation setup will come in a PR after the [third person controller PR](https://github.com/Unity-Technologies/com.unity.multiplayer.samples.bitesize/pull/66) is merged into develop.

### Pictures

**Overall Level:**

image
image

**Ingredients:**

image
image
image

**Stoves:**

image
image

**Drop Zones:**

image
image

### CODE REVIEW

File: Basic/ClientDriven/Packages/manifest.json [hunk 1]
```diff
@@ -7,7 +7,8 @@
     "com.unity.ide.vscode": "1.2.5",
     "com.unity.inputsystem": "1.4.3",
     "com.unity.multiplayer.samples.coop": "https://github.com/Unity-Technologies/com.unity.multiplayer.samples.coop.git?path=/Packages/com.unity.multiplayer.samples.coop",
-    "com.unity.netcode.gameobjects": "1.1.0",
+    "com.unity.netcode.gameobjects": "1.0.1",
```
Comment (RikuTheFuffs) [inline Basic/ClientDriven/Packages/manifest.json:10]: is this NGO downgrade intentional?

Comment (jilfranco-unity) [reply Basic/ClientDriven/Packages/manifest.json]: No that should be fixed now! :)

**Comment by SamuelBellomo** [comment]
Great job on this, this looks amazing!
Nit picking: The purple ingredients seem to have a weird contrast on the purple background. Dunno if we should play with an additional color in there, to add more contrast?

**Comment by jilfranco-unity** [comment]
> Great job on this, this looks amazing! Nit picking: The purple ingredients seem to have a weird contrast on the purple background. Dunno if we should play with an additional color in there, to add more contrast?

Good shout yeah! I've updated the materials to pop a bit more from the background, going for more of an "unflavored" look for the purple ingredients. Lemme know what you think! :)


image
image
image

**Comment by chrispope** [comment]
Seeing a warning when testing on Mac:
`Metal: Vertex shader missing buffer binding at index 0 (UnityPerDraw / Hidden/Internal-Loading)`

I can not pinpoint it to any material/shader. Do we know what it means or what impact it has on the project?

**Comment by jilfranco-unity** [comment]
> Seeing a warning when testing on Mac: `Metal: Vertex shader missing buffer binding at index 0 (UnityPerDraw / Hidden/Internal-Loading)`
> 
> I can not pinpoint it to any material/shader. Do we know what it means or what impact it has on the project?

This is actually a [known issue with ShaderGraph fixed in Unity 2022.2](https://issuetracker.unity3d.com/issues/metal-vertex-shader-missing-buffer-binding-at-index-warnings-after-saving-new-vfx-graph)!

File: Basic/ClientDriven/Assets/Scripts/Dropzone.cs [hunk 2]
```diff
@@ -1,10 +1,23 @@
+using System;
 using UnityEngine;
+using Random = UnityEngine.Random;
```
Comment (fernando-cortez) [inline Basic/ClientDriven/Assets/Scripts/Dropzone.cs:3]: Just a comment to remove this if it's not necessary.

File: Basic/ClientDriven/Assets/Shaders/SG_Pickups.shadergraph [hunk 3]
```diff
@@ -0,0 +1,4716 @@
+{
```
Comment (fernando-cortez) [inline Basic/ClientDriven/Assets/Shaders/SG_Pickups.shadergraph:1]: Seeing the following shader warning in the inspector for this shader, as well as SG_SimpleLit & UrpTriplanar:
'UnityMetaVertexPosition': implicit truncation of vector type

**Comment by jilfranco-unity** [comment]
> Just to confirm, that known issue relates to the VFX graph, not Shader Graph, are we sure it is the same known issue?

Ah I see, yes this is a different issue. I believe it's caused by some materials still trying to use Built-In shaders instead of URP ones. I've updated them to use the URP shaders now--please let me know if you still see this issue! @chrispope

File: Basic/ClientDriven/Assets/Shaders/SG_Pickups.shadergraph (hunk 3 above)
Comment (jilfranco-unity) [reply Basic/ClientDriven/Assets/Shaders/SG_Pickups.shadergraph:1]: I wasn't seeing the errors in the inspector on my side for some reason, but I believe I've fixed this issue on these shaders by explicitly truncating those vectors from vector4s to vector3s! please let me know if you still see the issue :)

Comment (fernando-cortez) [reply Basic/ClientDriven/Assets/Shaders/SG_Pickups.shadergraph:1]: Not seeing them anymore 👍

**Review by fernando-cortez — APPROVED** [review]

**Pull request merged**